Assets/DanmakU/Runtime/DanmakuState.cs

```diff
@@ -37,6 +38,7 @@ public struct DanmakuConfig {
 [Serializable]
 public struct DanmakuState {
   public Vector2 Position;
+  public Vector2 OldPosition;
```

Comment:
Ditto'ing the sentiment here. OldPositions shouldn't be exposed publicly, which sort of damages the POCO status of DanmakuState as well, should it's access modifier be changed.

For the most part, since DanmakuState is mostly used to initialize new bullets being fired, perhaps short circuiting it to just use Position as the OldPosition would be a wise change: using OldPosition == Position creates a static bullet that has not moved for a tick, a safe assumption about new bullets that have just been fired.

Reply:
Removed from the DanmakuState object.